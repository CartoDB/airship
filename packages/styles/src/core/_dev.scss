@import 'tools/_mixins';
@import 'variables/_variables';
