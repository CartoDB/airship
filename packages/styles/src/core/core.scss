@import 'variables/_variables';

* {
  box-sizing: border-box;
  font-family: sans-serif;
}

:root {
  --primary-color: #{$primary-color};
}
