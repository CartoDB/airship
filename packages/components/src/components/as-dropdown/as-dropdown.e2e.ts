import { E2EPage, newE2EPage } from '@stencil/core/testing';

describe('as-category-widget', async () => {
  describe('Rendering', async () => {
    let page: E2EPage;

    beforeEach(async () => {
      page = await newE2EPage();

      await page.setContent(`
        <as-dropdown></as-dropdown>
      `);
    });

    it('should render properly', async () => {
      const element = await page.find('as-dropdown');
      element.setProperty('options', exampleOptions);
      element.setProperty('selectedOption', 'option2');

      await page.waitForChanges();

      expect(element.outerHTML).toMatchSnapshot();
    });

    it('should render clear button when showClearButton is true and an option is selected', async () => {
      const element = await page.find('as-dropdown');
      element.setProperty('options', exampleOptions);
      element.setProperty('selectedOption', 'option2');
      element.setProperty('showClearButton', true);

      await page.waitForChanges();

      expect(element.outerHTML).toMatchSnapshot();
    });

    it('should show defaultText when there is no selected option', async () => {
      const element = await page.find('as-dropdown');
      element.setProperty('options', exampleOptions);
      element.setProperty('defaultText', 'Dropdown');

      await page.waitForChanges();

      expect(element.outerHTML).toMatchSnapshot();
    });
  });

  describe('Behaviour', async () => {
    let page: E2EPage;

    beforeEach(async () => {
      page = await newE2EPage();

      await page.setContent(`
        <as-dropdown></as-dropdown>
      `);
    });

    it('should emit an event containing selected option when option changes', async () => {
      const element = await page.find('as-dropdown');
      element.setProperty('options', exampleOptions);
      await page.waitForChanges();

      // Open Dropdown
      await page.click('.as-dropdown__control');
      await page.waitForChanges();

      const categoriesSelectedSpy = await page.spyOnEvent('optionChanged');

      await page.click('.as-dropdown__list-item button');
      await page.waitForChanges();

      expect(categoriesSelectedSpy).toHaveReceivedEventDetail(exampleOptions[0].value);
    });

    it('should open dropdown when control is clicked', async () => {
      const element = await page.find('as-dropdown');
      element.setProperty('options', exampleOptions);

      await page.waitForChanges();

      await page.click('.as-dropdown__control');
      await page.waitForChanges();

      const menuDropdown = await page.find('.as-dropdown');
      expect(menuDropdown).toHaveClass('as-dropdown--open');
    });

    it('should clear option when clear button is pressed', async () => {
      const element = await page.find('as-dropdown');
      element.setProperty('options', exampleOptions);
      element.setProperty('selectedOption', 'option2');
      element.setProperty('showClearButton', true);

      await page.waitForChanges();
      await page.click('.as-dropdown__clear');
      await page.waitForChanges();

      const selectedOption = await element.getProperty('selectedOption');
      expect(selectedOption).toBeNull();
    });
  });
});

const exampleOptions = [
  { text: 'Option 1', value: 'option1' },
  { text: 'Option 2', value: 'option2' },
  { text: 'Option 3', value: 'option3' }
];
