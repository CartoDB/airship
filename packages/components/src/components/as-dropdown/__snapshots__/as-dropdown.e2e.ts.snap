// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`as-dropdown-widget Rendering should render clear button when showClearButton is true and an option is selected 1`] = `"<as-dropdown class=\\"hydrated\\"><div class=\\"as-dropdown as-dropdown--clear\\"><button class=\\"as-dropdown__control as-body\\" aria-haspopup=\\"true\\">Option 2<div class=\\"as-dropdown__arrow\\"><svg viewbox=\\"2 6 12 6\\" xmlns=\\"http://www.w3.org/2000/svg\\"><path d=\\"M8.30988427,11.1237193 L7.69011573,11.1237193 L15.2195214,4.12372525 C15.415115,3.94188438 15.7124148,3.96294311 15.8835591,4.17076125 C16.0547035,4.37857939 16.0348835,4.69446043 15.83929,4.8763013 L8.30988427,11.8762953 C8.13246042,12.041244 7.86753958,12.041244 7.69011573,11.8762953 L0.160710032,4.8763013 C-0.034883519,4.69446043 -0.0547035068,4.37857939 0.116440851,4.17076125 C0.287585208,3.96294311 0.584885024,3.94188438 0.780478575,4.12372525 L8.30988427,11.1237193 Z\\"></path></svg></div></button><ul tabindex=\\"-1\\" class=\\"as-dropdown__list\\"><li class=\\"as-dropdown__list-item\\" data-value=\\"option1\\"><button class=\\"as-body\\">Option 1</button></li><li class=\\"as-dropdown__list-item\\" data-value=\\"option2\\"><button class=\\"as-body is-selected\\">Option 2</button></li><li class=\\"as-dropdown__list-item\\" data-value=\\"option3\\"><button class=\\"as-body\\">Option 3</button></li></ul><button class=\\"as-dropdown__clear\\"><svg viewbox=\\"2 2 12 12\\" xmlns=\\"http://www.w3.org/2000/svg\\"><path d=\\"M8,8.58232323 L1.70292632,14.8793969 C1.5421222,15.040201 1.28140721,15.040201 1.12060309,14.8793969 C0.95979897,14.7185928 0.95979897,14.4578778 1.12060309,14.2970737 L7.41767677,8 L1.12060309,1.70292632 C0.95979897,1.5421222 0.95979897,1.28140721 1.12060309,1.12060309 C1.28140721,0.95979897 1.5421222,0.95979897 1.70292632,1.12060309 L8,7.41767677 L14.2970737,1.12060309 C14.4578778,0.95979897 14.7185928,0.95979897 14.8793969,1.12060309 C15.040201,1.28140721 15.040201,1.5421222 14.8793969,1.70292632 L8.58232323,8 L14.8793969,14.2970737 C15.040201,14.4578778 15.040201,14.7185928 14.8793969,14.8793969 C14.7185928,15.040201 14.4578778,15.040201 14.2970737,14.8793969 L8,8.58232323 Z\\"></path></svg></button></div></as-dropdown>"`;

exports[`as-dropdown-widget Rendering should render properly 1`] = `"<as-dropdown class=\\"hydrated\\"><div class=\\"as-dropdown\\"><button class=\\"as-dropdown__control as-body\\" aria-haspopup=\\"true\\">Option 2<div class=\\"as-dropdown__arrow\\"><svg viewbox=\\"2 6 12 6\\" xmlns=\\"http://www.w3.org/2000/svg\\"><path d=\\"M8.30988427,11.1237193 L7.69011573,11.1237193 L15.2195214,4.12372525 C15.415115,3.94188438 15.7124148,3.96294311 15.8835591,4.17076125 C16.0547035,4.37857939 16.0348835,4.69446043 15.83929,4.8763013 L8.30988427,11.8762953 C8.13246042,12.041244 7.86753958,12.041244 7.69011573,11.8762953 L0.160710032,4.8763013 C-0.034883519,4.69446043 -0.0547035068,4.37857939 0.116440851,4.17076125 C0.287585208,3.96294311 0.584885024,3.94188438 0.780478575,4.12372525 L8.30988427,11.1237193 Z\\"></path></svg></div></button><ul tabindex=\\"-1\\" class=\\"as-dropdown__list\\"><li class=\\"as-dropdown__list-item\\" data-value=\\"option1\\"><button class=\\"as-body\\">Option 1</button></li><li class=\\"as-dropdown__list-item\\" data-value=\\"option2\\"><button class=\\"as-body is-selected\\">Option 2</button></li><li class=\\"as-dropdown__list-item\\" data-value=\\"option3\\"><button class=\\"as-body\\">Option 3</button></li></ul></div></as-dropdown>"`;

exports[`as-dropdown-widget Rendering should show defaultText when there is no selected option 1`] = `"<as-dropdown class=\\"hydrated\\"><div class=\\"as-dropdown\\"><button class=\\"as-dropdown__control as-body\\" aria-haspopup=\\"true\\">Dropdown<div class=\\"as-dropdown__arrow\\"><svg viewbox=\\"2 6 12 6\\" xmlns=\\"http://www.w3.org/2000/svg\\"><path d=\\"M8.30988427,11.1237193 L7.69011573,11.1237193 L15.2195214,4.12372525 C15.415115,3.94188438 15.7124148,3.96294311 15.8835591,4.17076125 C16.0547035,4.37857939 16.0348835,4.69446043 15.83929,4.8763013 L8.30988427,11.8762953 C8.13246042,12.041244 7.86753958,12.041244 7.69011573,11.8762953 L0.160710032,4.8763013 C-0.034883519,4.69446043 -0.0547035068,4.37857939 0.116440851,4.17076125 C0.287585208,3.96294311 0.584885024,3.94188438 0.780478575,4.12372525 L8.30988427,11.1237193 Z\\"></path></svg></div></button><ul tabindex=\\"-1\\" class=\\"as-dropdown__list\\"><li class=\\"as-dropdown__list-item\\" data-value=\\"option1\\"><button class=\\"as-body\\">Option 1</button></li><li class=\\"as-dropdown__list-item\\" data-value=\\"option2\\"><button class=\\"as-body\\">Option 2</button></li><li class=\\"as-dropdown__list-item\\" data-value=\\"option3\\"><button class=\\"as-body\\">Option 3</button></li></ul></div></as-dropdown>"`;
